### PR TITLE
spacy: Fix build

### PR DESCRIPTION
**Pull request by danieldk**
###### Motivation for this change

Fix `thinc` and `spacy` builds.

ZHF #45960

To make spacy usable again, the model packages need to be updated as well. I will do this in a separate PR if I have time later today or tomorrow.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/development/python-modules/spacy/default.nix [hunk 1]
```diff
@@ -37,7 +37,8 @@ buildPythonPackage rec {
       --replace "ftfy==" "ftfy>=" \
       --replace "msgpack-python==" "msgpack-python>=" \
       --replace "msgpack-numpy==" "msgpack-numpy>=" \
-      --replace "pathlib" "pathlib; python_version<\"3.4\""
+      --replace "thinc>=6.10.3,<6.11.0" "thinc>=6.10.3,<6.12.0" \
+      --replace "plac<1.0.0,>=0.9.6" "plac<1.1.0,>=0.9.6"
```
Comment (Mic92) [inline pkgs/development/python-modules/spacy/default.nix]: Why not removing the upper bound?

Comment (danieldk) [reply pkgs/development/python-modules/spacy/default.nix]: I could do that, but I am not sure how much these packages stick to semver or any API stability for that matter ;). I did some small tests to confirm that plac 1.0 and thinc 6.11 work with spacy.

But I'll update the PR without these upper bounds.

**Comment by danieldk** [comment]
I have removed the upper bounds.

**Comment by Mic92** [comment]
backport:

[detached HEAD 7888e2602e3] thinc: relax dependency version requirements
 Author: Daniël de Kok <me@danieldk.eu>
 Date: Wed Sep 5 08:17:34 2018 +0200
 1 file changed, 3 insertions(+), 1 deletion(-)
[detached HEAD e928fd9f33d] spacy: relax dependency versions
 Author: Daniël de Kok <me@danieldk.eu>
 Date: Wed Sep 5 08:38:52 2018 +0200
 1 file changed, 2 insertions(+), 1 deletion(-)

**Pull request merged**